Scripts/Runtime/Entities/Lifecycle/Spawner/EntityCommandBufferWithID.cs

```diff
@@ -0,0 +1,152 @@
+using Anvil.CSharp.Data;
+using System;
+using System.Collections.Generic;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    public struct EntityCommandBufferWithID
+    {
+        public static uint GetPersistentID()
+        {
+            return EntityCommandBufferWithIDManager.GetNextID();
+        }
+
+        public static void ClearPreviousInstanceIfExists(uint id)
+        {
+            EntityCommandBufferWithIDManager.Unregister(id);
+        }
+        
+
+        public readonly uint ID;
+
+        [NativeDisableContainerSafetyRestriction] [NativeDisableUnsafePtrRestriction]
+        private EntityCommandBuffer m_EntityCommandBuffer;
+
+        public EntityCommandBufferWithID(uint id, EntityCommandBuffer entityCommandBuffer)
```

Comment:
Can we have this created within this struct? Since we've disabled safety we should try to guarantee that the provided ECB isn't used elsewhere.

Maybe just pass in an activation delegate so the consumer can define where the ECB comes from.

Reply:
Done. 

We take in the `EntityCommandBufferSystem` if needed. We'll use it or we'll just create you a new one for temp usage.